### ISSUE TITLE
Guidance on Infra Deployment Pipeline

### ISSUE THREAD

**Issue by yradsmikham** [microsoft/bedrock#843]
**As a:**
Operator

**I want:**
to have complete documentation to create an end-to-end CI/CD infrastructure workflow using `spk` (in Azure DevOps)

**So that:**
I can automate any deployments and/or updates made to my infrastructure according to best practices.

**Describe the solution you'd like:**
A README that will provide guidance around constructing a workflow for deploying infrastructure to production.

- [x] Infrastructure Deployment Pipeline README: 
    - [x] Discuss different approaches to achieving infrastructure deployment securely.
    - [x] Update the original [SPK Infra end-to-end doc](https://github.com/CatalystCode/spk/blob/master/docs/infra/spk-infra-end-to-end.md).
    - [x] Update the original [SPK "under the hood" doc](https://github.com/CatalystCode/spk/blob/master/docs/infra/spk-infra-under-the-hood.md).

**Does this require updates to documentation?:**
Yes

**Comment by andrebriggs** [comment]
@yradsmikham @NathanielRose is a new `spk infra` command going to generate a azure-pipeline.yaml file that will represent a pipeline similar to what `spk services` commands do?  If so how will an end user get that that?

**Comment by yradsmikham** [comment]
@andrebriggs for now, no. This could be an enhancement in the future. `spk infra scaffold` and `spk infra generate` should mature first. For now, there is no harm in manually creating `azure-pipeline.yaml` files.

**Comment by andrebriggs** [comment]
@yradsmikham @NathanielRose I want to make sure that manually creating an azure-pipeline.yaml file will benefit the end user. Are you suggesting that we create a static file? If so how will the end user "get" that?

> spk infra scaffold and spk infra generate should mature first.

It sounds like maybe we should wait on the pipeline since it will basically be a POC?

**Comment by yradsmikham** [comment]
@andrebriggs sure, that works!

**Comment by andrebriggs** [comment]
@yradsmikham @NathanielRose so we won't do this task right now. What tasks need to get done before you think we can start thinking about this?

**Comment by yradsmikham** [comment]
@andrebriggs let's focus on one pipeline at a time. It might make sense to have the [Generation Pipeline](https://github.com/microsoft/bedrock/issues/827) proofed out first.

**Comment by NathanielRose** [comment]
What we could do for an E2E workflow Integration Test could look similar to this:
- Create a pipeline after infra has been generated in a repo using spk `az pipelines create -n $PIPELINE_NAME -project -repo`.
- Use a pregenerated simplified version of our [bedrock pipeline yaml](https://github.com/microsoft/bedrock/blob/master/azure-pipelines.yml) inside the validation repo.
- Have a `go.mod` and terratest go script to validate the simple infra.

**Comment by yradsmikham** [comment]
From our discussion, it appears that we may not want a "concrete" solution for how to create a deployment pipeline. Instead, we will write up a brief doc on how to approach it as there are many ways to go about this.
